### ISSUE TITLE
How to force electron-updater to update to the x64 package on M1 Macs?

### ISSUE THREAD

**Issue by rathboma** [electron-userland/electron-builder#6851]
Hey folks,

Firstly, and as always -- thank you for maintaining electron-builder. I couldn't make an app solo without all of your amazing work.

I have a problem in that one of my dependencies (oracle's database instant client - c++ library) is not available for ARM64, only for x86_64.

I can deal with this in the app, but I need to have a way to make sure that users who install the x86 version can upgrade to the newest x86 version...even if they're on an M1 Mac and using Rosetta.

Right now the default behavior is to upgrade to the ARM64 version, which totally makes sense, except in this specific circumstance.

Ideally I'd like a way to override, this, is it possible? I looked at examples, but wasn't sure how to override the architecture for AutoUpdater

**Comment by mmaietta** [comment]
Hi there!
If there's no arm64 version available for download, I was under the impression that electron-updater would simply fall back to the default (x64).  It was implemented as part of https://github.com/electron-userland/electron-builder/pull/5524
It was to prevent users from being stuck on an arm64 version until another arm64 update was distributed.

**Comment by rathboma** [comment]
Yes correct! However I still want to publish an ARM64 version alongside my x86 version. 

I need M1 users, who are using the x86 version of the app to be able to continue using that version, rather than being upgraded to the ARM version, when both are available

**Comment by mmaietta** [comment]
Hmmm, in that case, I think a more simple solution is to create the update file for your purposes. What if we compile both binaries, but split in to separate steps? The 2nd run will generate the `latest-mac.yml` and overwrite the previously generated file, so `latest-mac.yml` would only contain details for x64.
```
electron-builder --mac --arm64
electron-builder --mac --x64
```

In terms of the PR, I'm honestly not a fan of adding a specific macOS-only flag to the base updater class when it's only applicable to arm64 builds.
Instead, I'm wondering if creating a callback like `shouldDownloadUpdate` could be a universal approach for allowing the host application to determine if the artifact satisfies certain conditions.

**Comment by mltucker** [comment]
@mmaietta The problem with this solution is that it would update all arm users to the latest x64 version, correct? The goal is to have arm binaries upgrade to arm and x64 binaries upgrade to x64.

**Comment by mltucker** [comment]
Hey @mmaietta, did you figure this out? My solution was to copy the package.json to package-arm.json during build, then replace the s3 bucket with a special arm s3 bucket. Then if you have the arm build you update to the latest arm build and if you're on the intel build you update to the latest intel build.
